src/canvas/Exporter.tsx

```diff
@@ -0,0 +1,171 @@
+            // TODO adjust length of lines before they are clipped.
+            // Clone original piece and scale.
+            const originalPatternPiece = patternPiece.clone();
+            this._scale(originalPatternPiece, inchesPerPixel);
+
+            // Calculate the number of pages in a row and in a column
+            const boundBox = originalPatternPiece.getBoundingBox();
+            const numPagesX = Math.ceil(boundBox.width / pageSizeX);
+            const numPagesY = Math.ceil(boundBox.height / pageSizeY);
+            
+            let positionX = 0;
+            let positionY = 0;
+            for (let x = 0; x < numPagesX; x++) {
+                positionX = x * pageSizeX;
+            
+                for (let y = 0; y < numPagesY; y++) {
+                    positionY = y * pageSizeY;
+                    this.doc?.addPage();
+                    console.log("page num ", this.doc?.getNumberOfPages());
+
+                    if (!ctx) {
+                        throw new Error("Could not create 2D context for canvas.");
+                    }
+
+                    ctx.clearRect(0, 0, pageSizeX, pageSizeY);
+
+                    // Draw white background and border
+                    ctx.fillStyle = 'white';
+                    ctx.fillRect(0, 0, pageSizeX, pageSizeY);
+                    ctx.strokeRect(0, 0, pageSizeX, pageSizeY);
+
+                    // Push a clip rect.
+                    ctx.save();
+                    ctx.rect(0, 0, pageSizeX, pageSizeY);
+                    ctx.clip();
+                    
+                    // Translate the piece, so that the correct section is drawn in the clip rect.
+                    const translatedPatternPiece = originalPatternPiece.clone();
+                    translatedPatternPiece.translate(new Vector(-positionX, -positionY));
+
+                    translatedPatternPiece.getAllPaths().forEach(patternPath => {
+                        ctx.stroke(patternPath.getPath2D());
+                    });
+            
+                    // Pop the clip rect.
+                    ctx.restore();
+
+                    // Create an image from the canvas and add it to the pdf.
+                    // Use JPEG intead of PNG because the JSPDF PNG encoder is slow and creates large documents.
+                    this.doc?.addImage(canvas, 'JPEG', margin, margin, pageWidth - (2 * margin), pageHeight - (2 * margin));
+                    this._printFooter(x + 1, y + 1);
+                }
+            }
+        });
+        
+        this.doc.save("test.pdf");
+    };
+
+    private _scale = (patternPiece: PatternPiece, scalar: number): void => {
+        patternPiece.scale(scalar);
+
+        const boundBox = patternPiece.getBoundingBox();
+        patternPiece.translate(new Vector(-boundBox.minX, -boundBox.minY));
+    };
+
+    private _printFooter = (x: number, y: number): void => {
+        this.doc?.text('Page (' + x + ', ' + y +')', .6, 10.35);
```

Comment:
Maybe add something to identify the piece? like "Piece no. 2, page (1, 2)" or "page (1, 2) of piece no. 2"

Reply:
I added the piece number, so now it reads like "piece 1, page (1, 3)". What do you think? I can't decide if all lower caps looks better, or if I should capitalize Piece.

Reply:
I think it looks fine either way, I have a slight preference for capitalizing piece.